Panos network operations hygine checks

adambaumeister opened the pull request:
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
This is the last of the configuration hygiene Checks that we use as part of the configuration hygiene use case.

These checks run through a variety of different configuration components to do with configuration visibility and we also provide some fix commands as well.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

GuyAfik reviewed (commented):
@adambaumeister great work! a few questions/notes
1) i see that in the main only 11 new commands were added while in the release-notes 13 were stated as added.
2) as usual, make sure there aren't any secrets in README.
3) also make sure you handle the build issues.

adambaumeister commented on Packs/PAN-OS/Integrations/Panorama/Panorama.py:
```diff
@@ -8721,6 +8725,172 @@ class ConfigurationHygieneCheckResult:
+        return result
+
+    @staticmethod
+    def fix_security_rule_security_profile_group(topology: Topology,
```
Very true - while the fix is differnet the way we get the rules is always the same. I've split this into it's own function.

adambaumeister commented on Packs/PAN-OS/Integrations/Panorama/Panorama.py:
```diff
@@ -8991,6 +9160,318 @@ def check_vulnerability_profiles(
+        return result
+
+    @staticmethod
+    def get_all_conforming_vulnerability_profiles(
```
We may add more checks into these two functions later -right now they are similar because the profiles have a similar object structure but there are some key differences that we don't currently action. Easier to read to keep them seperate I think.

adambaumeister commented on Packs/PAN-OS/Integrations/Panorama/README.md:
```diff
@@ -6138,3 +6138,799 @@ Install the given software version onto the device. Download the software first
+                {
+                    "container_name": "LAB",
+                    "description": "Log forwarding profile missing log type 'threat'.",
+                    "hostid": "192.168.1.145",
```
That IP is a private IP in my testlab - it's not a secret and it can stay in there (it's in the other command examples already).

adambaumeister commented:
@GuyAfik re the commands;

I mistakenly copied all the hygiene commands, forgetting I had already added two in the previous PR. I've corrected the release notes. After this, we only have 5 commands outstanding to integrate into this pack. The build issues should also be corrected now.

GuyAfik reviewed (commented):
Looks fantastic, last small comment and we setup a demo already.

adambaumeister commented:
* Look for better name for container_name in any panos objects
 * Rename PAN-OS object headers to more specific for conforming object commands
 * Re-order container_name

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=18915) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=18915) before we can accept your contribution.<br/>**5** out of **7** committers have signed the CLA.<br/><br/>:white_check_mark: julieschwartz18<br/>:white_check_mark: content-bot<br/>:white_check_mark: YuvHayun<br/>:white_check_mark: ostolero<br/>:white_check_mark: adambaumeister<br/>:x: AradCarmi<br/>:x: SarahFleising<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=18915) it.</sub>

adambaumeister commented:
Updated to fix the above stuff, dataclass_to_command_results now supports overriding the headers in the dataclass objects (useful when we have one dataclass that can be reused across many commands like PanosObject).

Also updated to re-order the output for the "conforming" commands.

GuyAfik commented on Packs/PAN-OS/Integrations/Panorama/Panorama.py:
```diff
@@ -8991,6 +9162,318 @@ def check_vulnerability_profiles(
+            security_rules: List[SecurityRule] = SecurityRule.refreshall(firewall_rulebase)
+            pre_security_rules: List[SecurityRule] = SecurityRule.refreshall(pre_rulebase)
+            post_security_rules: List[SecurityRule] = SecurityRule.refreshall(post_rulebase)
+            security_rules = security_rules + pre_security_rules + post_security_rules
```
@adambaumeister your input on that one?

adambaumeister replied:
Fixed!

GuyAfik reviewed (approved):
Looks awesome
